libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/services/ContentService.java

```diff
@@ -330,17 +340,55 @@ public Content add(Content entity) {
   public Content update(Content entity) {
     var result = repository.save(PrincipalHelper.updateAudit(entity));
 
+    // TODO: Shouldn't need to update all children. Need to find a better way to do
+    // this.
+    if (entity.getFileReferences().size() > 0) {
+      entity.getFileReferences().forEach((file) -> {
+        if (file.getId() == 0)
+          fileReferenceService.add(file);
+        else
+          fileReferenceService.update(file);
+      });
+    }
     if (entity.getContentActions().size() > 0) {
-      contentActionService.update(entity.getContentActions());
+      entity.getContentActions().forEach((action) -> {
+        if (action.getActionId() == 0)
+          contentActionService.add(action);
+        else
+          contentActionService.update(action);
+      });
     }
     if (entity.getContentTags().size() > 0) {
-      contentTagService.update(entity.getContentTags());
+      entity.getContentTags().forEach((tag) -> {
+        if (tag.getCreatedOn() == null) // TODO: I don't think this will work presently
```

Comment:
Need to update code to handle this scenario where we need `createdOn=null` so that we know it's a new record.